src/file-manager/fm-desktop-icon-view.c

```diff
@@ -350,7 +353,7 @@ fm_desktop_icon_view_handle_middle_click (CajaIconContainer *icon_container,
     }
 
     /* Stop the event because we don't want anyone else dealing with it. */
-    gdk_flush ();
+    gdk_display_flush (display);
     g_signal_stop_emission_by_name (icon_container, "middle_click");
```

Comment:
the same happens here, maybe is needed this before?

display = gtk_widget_get_display (GTK_WIDGET (icon_container));

Reply:
i agree, just updated the PR